resources/types/database.go

```diff
@@ -115,7 +115,17 @@ func getHost(resourceId string, state *output.TfState, cloud commonpb.CloudProvi
 		}
 		return values["address"].(string), nil
 	case commonpb.CloudProvider_AZURE:
-		values, err := state.GetValues(database.AzureMySqlServer{}, resourceId)
+		var azureDatabaseEngine database.AzureDatabaseEngine
+
+		if engine == resourcespb.DatabaseEngine_MYSQL {
+			azureDatabaseEngine = database.AzureMySqlServer{}
+		} else if engine == resourcespb.DatabaseEngine_POSTGRES {
+			azureDatabaseEngine = database.AzurePostgreSqlServer{}
+		} else if engine == resourcespb.DatabaseEngine_MARIADB {
+			azureDatabaseEngine = database.AzureMariaDbServer{}
+		}
```

Comment:
can you also add an else clause that throws just in case?

Reply:
what should be the statement in the else block?

Reply:
just return an error

return "", fmt.Errorf("unhandled engine %s", azureDatabaseEngine.String())